homeassistant/components/homematicip_cloud/climate.py

```diff
@@ -116,7 +119,11 @@ def preset_mode(self):
         if self._device.boostMode:
             return PRESET_BOOST
         if self._device.controlMode == HMIP_ECO_CM:
-            return PRESET_ECO
+            absence_type = self._home.get_functionalHome(IndoorClimateHome).absenceType
+            if absence_type == AbsenceType.VACATION:
+                return PRESET_AWAY
+            if absence_type in [AbsenceType.PERIOD, AbsenceType.PERMANENT]:
+                return PRESET_ECO
```

Comment:
Do we need an else condition to return a preset if it's not `VACATION`, `PERIOD`, or `PERMANENT` but still within the if block of `self._device.controlMode == HMIP_ECO_CM`?

Reply:
Good point. I added an else statement that should never be used(as far as i know!), but improve readability of the code. I also added AbsenceType.PARTY to the conditions for ECO mode.
Thanks.